### PR TITLE
SaveGame Fixes

### PR DESCRIPTION
**Pull request by fmatthew5876**
Depends on https://github.com/EasyRPG/liblcf/pull/277

List of variables we will have to get right in `Player`
* `SaveSystem::screen` 0x1
* `SavePartyLocation::processed` 0x4B
* `SaveMapEvent::stop_count` 0x34
* `SaveMapEvent::anim_count` 0x35

### CODE REVIEW

File: src/scene_save.cpp [hunk 1]
```diff
@@ -99,7 +99,10 @@ void Scene_Save::Action(int index) {
 	}
 
 	LSD_Reader::PrepareSave(Main_Data::game_data);
-	LSD_Reader::Save(filename, Main_Data::game_data, Player::encoding);
+	auto data_copy = LSD_Reader::ClearDefaults(Main_Data::game_data, Game_Map::GetMapInfo(), Game_Map::GetMap());
+	//RPT_RT saves always have the screen set to this.
```
Comment (fdelapena) [inline src/scene_save.cpp]:
```suggestion
	// RPG_RT saves always have the screen set to this.
```

Comment (fmatthew5876) [reply src/scene_save.cpp]: Fixed

File: src/scene_save.cpp [hunk 2]
```diff
@@ -99,7 +99,10 @@ void Scene_Save::Action(int index) {
 	}
 
 	LSD_Reader::PrepareSave(Main_Data::game_data);
-	LSD_Reader::Save(filename, Main_Data::game_data, Player::encoding);
+	auto data_copy = LSD_Reader::ClearDefaults(Main_Data::game_data, Game_Map::GetMapInfo(), Game_Map::GetMap());
+	// RPT_RT saves always have the screen set to this.
```
Comment (Ghabry) [inline src/scene_save.cpp]: RPT_RT

File: src/game_targets.cpp [hunk 3]
```diff
@@ -86,22 +86,21 @@ std::vector<RPG::SaveTarget*> Game_Targets::GetTeleportTargets() {
 	return targets;
 }
 
-void Game_Targets::SetEscapeTarget(int map_id, int x, int y, int switch_id) {
-	std::vector<RPG::SaveTarget>::iterator target = FindTarget(0, true);
+void Game_Targets::SetEscapeTarget(int map_id, int x, int y, bool switch_on, int switch_id) {
+	auto* target = &data[0];
```
[the remark this answers is not in the document]
Comment (fmatthew5876) [reply src/game_targets.cpp]: Better would be to store a direct object for escape and use a vector for teleports. But we don't have machinery in liblcf to do that yet.

File: src/game_interpreter.cpp [hunk 4]
```diff
@@ -2549,10 +2549,9 @@ bool Game_Interpreter::CommandTeleportTargets(RPG::EventCommand const& com) { //
 
 	int x = com.parameters[2];
 	int y = com.parameters[3];
-	int switch_id = (com.parameters[4] != 0)
-		? com.parameters[5]
-		: -1;
-	Game_Targets::AddTeleportTarget(map_id, x, y, switch_id);
+	bool switch_on = com.parameters[4];
```
Comment (Ghabry) [inline src/game_interpreter.cpp]: MSVC will complain that this is bad for performance because int is changed to bool without cast/condition

Comment (fmatthew5876) [reply src/game_interpreter.cpp]: Fixed

**Comment by Ghabry** [comment]
Jenkins: test this please

**Pull request merged**